lua/entities/gmod_wire_egp/lib/objects/wedge.lua

```diff
@@ -54,17 +54,19 @@ Obj.DataStreamInfo = function( self )
 	table.Merge( tbl, { angle = self.angle, size = self.size, fidelity = self.fidelity } )
 	return tbl
 end
-function Obj:Contains(point)
-	local theta = math.rad(point.angle + self.angle)
+function Obj:Contains(point, this)
+	local _, realpos = EGP:GetGlobalPos(this, self.index)
+	local x, y = (point.x - realpos.x) / self.w, (point.y - realpos.y) / self.h
```

Comment:
There should be a Obj:WorldToLocal function that takes a point and converts it to the obj's local coords. It would get rid of all the duplicate GetGlobalPos and cos/sin multiplication in these functions.

Reply:
Ideally
```lua
function Obj:Contains(x, y, this)
	x, y = self:WorldToLocal(x, y, this)
	x = x / self.w
	y = y / self.h
	if x * x + y * y > 1 then return false end
	theta = math.deg(math.atan2(y, x))
	return theta <= -self.size or 0 <= theta
end
```

Reply:
Hmm I swear I could've seen such a function before.

Reply:
Maybe also rename 'this' to 'egp'

Reply:
I don't believe having this function as a member of Obj is appropriate. They're not singletons, so I think it's better to put them as a member of EGP.

Reply:
Whatever works best. EGP is a bit of a mess.